SashaSales/src/main/java/ua/artcode/dao/DaoStudentImplJPA.java

```diff
@@ -2,121 +2,70 @@
     @Override
+    @Transactional
     public Student create(Student student) {
-        EntityManager entityManager = managerFactory.createEntityManager();
-        EntityTransaction entityTransaction = entityManager.getTransaction();
-
-        try {
-
-                entityTransaction.begin();
-                entityManager.persist(student);
-                entityTransaction.commit();
-
-        } catch (Exception e){
-            entityTransaction.rollback();
-            return null;
-        } finally {
-            entityManager.close();
-        }
 
+        entityManager.persist(student);
         return student;
     }
 
     @Override
+    @Transactional
     public boolean delete(String entity_name) {
-        EntityManager entityManager = managerFactory.createEntityManager();
-        EntityTransaction entityTransaction = entityManager.getTransaction();
 
         Student student = getEntityByName(entity_name, entityManager);
-
-        try {
-            entityTransaction.begin();
-            entityManager.remove(student);
-            entityTransaction.commit();
-        } catch (Exception e){
-            entityTransaction.rollback();
-            return false;
-        } finally {
-            entityManager.close();
-        }
+        entityManager.remove(student);
         return true;
     }
 
     @Override
+    @Transactional
     public Student updateByGroup(String student_name, String group_name) {
-        EntityManager entityManager = managerFactory.createEntityManager();
-        EntityTransaction entityTransaction = entityManager.getTransaction();
-        Student student = null;
-        Group group = null;
-
-        try {
-
-            entityTransaction.begin();
-
-            int idG = UtilsMethod.getIdOfGroup(group_name, managerFactory);
-            int idS = UtilsMethod.getIdOfStudent(student_name, managerFactory);
-            student = entityManager.find(Student.class, idS);
-            group = entityManager.find(Group.class, idG);
-            student.setGroup(group);
-            entityTransaction.commit();
-
-        } catch (Exception e){
-            entityTransaction.rollback();
-            return null;
-        } finally {
-            entityManager.close();
-        }
+
+        int idG = UtilsMethod.getIdOfGroup(group_name, entityManager);
+        int idS = UtilsMethod.getIdOfStudent(student_name, entityManager);
+        Student student = entityManager.find(Student.class, idS);
+        Group group = entityManager.find(Group.class, idG);
+        student.setGroup(group);
+
         return student;
     }
 
     @Override
     public Student findById(Object id) {
-        EntityManager entityManager = managerFactory.createEntityManager();
 
-        try {
-            return entityManager.find(Student.class, id);
-        } finally {
-            entityManager.close();
-        }
+        return entityManager.find(Student.class, id);
     }
 
     @Override
     public List<Student> getAll() {
```

Comment:
use pagination, dont retrieve all data per one request
add arguments as offset and limit